### ISSUE TITLE
📚 API : make sure the API remains Exojax compatible

### ISSUE THREAD

**Issue by erwanp** [radis/radis#597]
### 💭 Description

Since #480 Radis embeds a common API shared by Radis-Line-by-line  (spectrum calculation code) and [Exojax](https://secondearths.sakura.ne.jp/exojax/)

We should add tests in the Radis CI to make sure the API always remains fully compatible with Exojax functions (in terms of signature, i.e. input parmaters ). 

I see it as : 

- in the develop branch, add classes representatives of how Exojax uses the API ; and make sure it works 
- before each new merge to the Master branch, add in our Travis-CI test suite a full installation & test of Exojax  

@HajimeKawahara @YuiKasagi any idea / request on this ?

**Comment by HajimeKawahara** [comment]
This is a bit difficult problem but important for long time development.  Writing some test code for api.py in ExoJAX, and importing these tests to radis? In general, writing the unit tests for database I/O is a bit challenging because it sometimes needs to be downloaded from the database websites.

**Comment by erwanp** [comment]
Yes. Do you already have some unitests in Exojax/api? I can borrow them and add them in Radis.api to ensure backward compatibility

**Comment by HajimeKawahara** [comment]
back to this issue. See [exojax #462](https://github.com/HajimeKawahara/exojax/pull/462)
Maybe you are right. We need to make the unittest of ExoJAX/api in exojax?

**Comment by erwanp** [comment]
We do 2 things : 

- for some test cases, we are downloading small databases from Exojax (SiO for instance), which requires ExoMol to be online; but limits the data transfer
- for other test cases, we have mock-data available in https://github.com/radis/radis/tree/develop/radis/test/files  and that can be fetched from anywhere with [radis.test.utils.getTestFile](https://radis.readthedocs.io/en/latest/source/radis.test.utils.html#radis.test.utils.getTestFile)

```python
from radis.test.utils import getTestFile
from radis import load_spec
load_spec(getTestFile('CO_Tgas1500K_mole_fraction0.01.spec'))
```

These are very small files, shipped with Radis, usually used to ensure I/O compatibility

**Comment by HajimeKawahara** [comment]
Thanks. It looks there is no ExoMol fragment file available in [radis.test.utils.getTestFile](https://radis.readthedocs.io/en/latest/source/radis.test.utils.html#radis.test.utils.getTestFile). Then, I will use a fragment ExoMol file in exojax.

**Comment by minouHub** [comment]
@code2929563 you could also include a fragment of a Kurucz or NIST file that we always store in the `radis/db` folder. That would allow testing the loading and processing without downloading from the websites
